opensearch_py_ml/ml_commons/model_access_control.py

```diff
@@ -0,0 +1,99 @@
+    validate_create_model_group_parameters,
+    validate_delete_model_group_parameters,
+    validate_search_model_group_parameters,
+    validate_update_model_group_parameters,
+)
+
+
+class ModelAccessControl:
+    API_ENDPOINT = "model_groups"
+
+    def __init__(self, os_client: OpenSearch):
+        self.client = os_client
+
+    def register_model_group(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        access_mode: Optional[str] = "private",
+        backend_roles: Optional[List[str]] = None,
+        add_all_backend_roles: Optional[bool] = False,
+    ):
+        validate_create_model_group_parameters(
+            name, description, access_mode, backend_roles, add_all_backend_roles
+        )
+
+        body = {"name": name, "add_all_backend_roles": add_all_backend_roles}
+        if description:
+            body["description"] = description
+        if access_mode:
+            body["access_mode"] = access_mode
+        if backend_roles:
+            body["backend_roles"] = backend_roles
+
+        return self.client.transport.perform_request(
+            method="POST", url=f"{ML_BASE_URI}/{self.API_ENDPOINT}/_register", body=body
+        )
+
+    def update_model_group(
+        self,
+        update_query: dict,
+        model_group_id: Optional[str] = None,
+    ):
+        validate_update_model_group_parameters(update_query, model_group_id)
+        return self.client.transport.perform_request(
+            method="PUT",
+            url=f"{ML_BASE_URI}/{self.API_ENDPOINT}/{model_group_id}",
+            body=update_query,
+        )
+
+    def search_model_group(self, query: dict):
+        validate_search_model_group_parameters(query)
+        return self.client.transport.perform_request(
+            method="GET", url=f"{ML_BASE_URI}/{self.API_ENDPOINT}/_search", body=query
+        )
+
+    def search_model_group_by_name(self, model_group_name, _source=None, size=1):
+        query = {"query": {"match": {"name": model_group_name}}, "size": size}
+        if _source:
+            query["_source"] = _source
+        return self.search_model_group(query)
+
+    def get_model_group_id_by_name(self, model_group_name):
+        try:
+            res = self.search_model_group_by_name(model_group_name)
+            if res["hits"]["hits"]:
+                return res["hits"]["hits"][0]["_id"]
+            else:
+                return None
+        except NotFoundError:
+            return None
+
+    def delete_model_group(
+        self,
+        model_group_id: str = None,
+    ):
+        validate_delete_model_group_parameters(model_group_id)
+        return self.client.transport.perform_request(
+            method="DELETE", url=f"{ML_BASE_URI}/{self.API_ENDPOINT}/{model_group_id}"
+        )
+
+    def delete_model_group_by_name(self, model_group_name):
+        model_group_id = self.get_model_group_id_by_name(model_group_name)
+        if model_group_id is None:
```

Comment:
Should we send a error message for customer that model group isn't found?

Reply:
Sure. I think we can raise a NotFoundError here. I will add that to the code.

Reply:
Although, I created it solely for the purpose of using it in texts. Can i add an argument `safe_delete` which won't raise error? Otherwise, i need to add lot of try.. except blocks just to handle model group deletions.

Reply:
If that's a public function, we don't want any parameter which won't make enough sense for customers?

Can customer get benefited from `safe_delete`. What will `safe_delete` do?

Reply:
@dhrubo-os , I think making this method a private method makes sense. Since, you mentioned, using name as primary key isn't the actual way right now. What do you think?

Reply:
I fixed this. Updated tests to catch the exceptions.